src/django_oapif/views.py

```diff
@@ -80,7 +82,12 @@ def get(self, request, *args, **kwargs):
             # Instantiating the viewset
             viewset = Viewset(request=request)
 
-            collections.append(viewset._describe(request, base_url=f"collections/"))
+            # Returning only viewsets that match the request's permissions
+            if all(
+                permission.has_permission(request, self)
+                for permission in viewset.get_permissions()
+            ):
```

Comment:
Just thinking, are you sure it's supposed to be "all" and not "any" ? Doesn't matching one permission suffice in DRF ?

Actually, we shouldn't implement it ourselves, especially since this is quite a sensitive topic. I think you can use `viewset.check_permissions(request)` (defined on the APIView)  and catch `exceptions.NotAuthenticated` and `exceptions.PermissionDenied`